data-runtime/src/main/java/io/micronaut/data/runtime/operations/internal/AbstractSyncEntitiesOperations.java

```diff
@@ -157,7 +158,7 @@ protected void triggerPost(Consumer<EntityEventContext<Object>> fn) {
 
     @Override
     public List<T> getEntities() {
-        return entities.stream().map(d -> d.entity).toList();
+        return entities.stream().map(d -> d.entity).collect(Collectors.toList());
```

Comment:
This was returning unmodifiable collection but in cascade operations we are updating items in the iterator so need to revert change done in https://github.com/micronaut-projects/micronaut-data/pull/2905/files#diff-8a51dce18e47b3d23f3c13d9e60b50738f370fd0e2ca14db87cadac7c4303da5R160